src/imguiclient/imguiapp.cpp

```diff
@@ -1410,66 +1404,77 @@ std::string ImGuiApp::create_time_string(std::chrono::time_point<std::chrono::sy
 
 void ImGuiApp::BackgroundWindow()
 {
-    // #TODO: Make the background dockable somehow. There is a example in ImGui Demo > Examples > Dockspace
+    // Configure flags for the background window
+    ImGuiWindowFlags window_flags = ImGuiWindowFlags_MenuBar | ImGuiWindowFlags_NoDocking | ImGuiWindowFlags_NoTitleBar
+        | ImGuiWindowFlags_NoCollapse | ImGuiWindowFlags_NoResize | ImGuiWindowFlags_NoMove
+        | ImGuiWindowFlags_NoBringToFrontOnFocus | ImGuiWindowFlags_NoNavFocus;
+
+    // Get the main viewport
+    const ImGuiViewport *viewport = ImGui::GetMainViewport();
+
+    // Set window position and size to cover the entire viewport
+    ImGui::SetNextWindowPos(viewport->WorkPos);
+    ImGui::SetNextWindowSize(viewport->WorkSize);
+    ImGui::SetNextWindowViewport(viewport->ID);
 
-    // clang-format off
-    constexpr ImGuiWindowFlags window_flags =
-        ImGuiWindowFlags_NoDecoration |
-        ImGuiWindowFlags_NoMove |
-        ImGuiWindowFlags_NoBackground |
-        ImGuiWindowFlags_MenuBar |
-        ImGuiWindowFlags_NoBringToFrontOnFocus |
-        ImGuiWindowFlags_NoDocking;
-    // clang-format on
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowRounding, 0.0f);
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowBorderSize, 0.0f);
 
+    // Push zero padding to maximize space
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowPadding, ImVec2(0.0f, 0.0f));
+
+    // Begin the background window
     bool window_open = true;
-    ImScoped::Window window("main", &window_open, window_flags);
-    if (window.IsContentVisible)
+    ImGui::Begin("DockSpace", &window_open, window_flags);
+
+    // Pop the style variables we pushed
+    ImGui::PopStyleVar(3);
+
+    // Submit the DockSpace
+    ImGuiIO &io = ImGui::GetIO();
+    if (io.ConfigFlags & ImGuiConfigFlags_DockingEnable)
     {
-        ImGui::SetWindowPos("main", m_windowPos);
-        ImGui::SetWindowSize("main", ImVec2(m_windowSize.x, 0.f));
+        ImGuiID dockspace_id = ImGui::GetID("MyDockSpace");
```

Comment:
Can put a better name for the Dockspace. Maybe "DockSpaceId"

Reply:
I meant the string, not the variable name.

Reply:
Done